Logger/FullName.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Text;
+
+namespace Logger;
+
+// value type, since it is more helpful to check equality of a name's data (the actual name) than its reference (the location in memory)
+// not immutable, as someone's name can change
+public record class FullName
```

Comment:
```suggestion
public record struct FullName
```
 I think a record struct could of been more appropriate here